Gems/LyShine/Code/Source/UiCanvasComponent.cpp

```diff
@@ -1196,7 +1199,35 @@ bool UiCanvasComponent::HandleInputEvent(const AzFramework::InputChannel::Snapsh
     {
         if (!m_renderToTexture && m_isPositionalInputSupported)
         {
+#if defined(CARBONATED)
+            float viewportWidth;
+            float viewportHeight;
+
+            if (viewportSize)
+            {
+                viewportWidth = viewportSize->GetX();
+                viewportHeight = viewportSize->GetY();
+            }
+            else
+            {
+                UiRenderer* uiRenderer = m_renderInEditor ? GetUiRendererForEditor() : GetUiRendererForGame();
+                AZ_Assert(uiRenderer, "UiRenderer is not available");
+
+                const AZ::Vector2 renderViewportSize = uiRenderer->GetViewportSize();
```

Comment:
I wonder if it returns the screen resolution or the render resolution. You can test it by settings r_renderScale less than 1, then checking what is the result for viewportWidth & viewportHeight. We might have another issue here. Please test it and let me know. You can do it in runtime I believe by running console command r_renderScale = 0.8 or adding it to a setreg file.

Reply:
I saw that with the world map and the game HUD UI the user input touch event comes in screen coordinates, so in theory this should return screen, NOT the reduced render.